File: python/file_operations.py
"""
File I/O utility functions for test automation
"""

import os
import pandas as pd
from pandas import DataFrame
from typing import List


def read_text_file(file_path: str) -> str:
    """
    Read a text file and return its contents
    
    Args:
        file_path: Path to the text file
        
    Returns:
        File contents as a string
    """
    with open(file_path, 'r', encoding='utf-8') as file:
        return file.read()


def read_text_file_lines(file_path: str) -> List[str]:
    """
    Read a text file and return its contents as a list of lines
    
    Args:
        file_path: Path to the text file
        
    Returns:
        List of lines from the file
    """
    with open(file_path, 'r', encoding='utf-8') as file:
        return file.readlines()


def write_text_file(file_path: str, content: str) -> None:
    """
    Write content to a text file
    
    Args:
        file_path: Path to the text file
        content: Content to write
    """
    with open(file_path, 'w', encoding='utf-8') as file:
        file.write(content)


def append_to_text_file(file_path: str, content: str) -> None:
    """
    Append content to a text file
    
    Args:
        file_path: Path to the text file
        content: Content to append
    """
    with open(file_path, 'a', encoding='utf-8') as file:
        file.write(content)


def file_exists(file_path: str) -> bool:
    """
    Check if a file exists
    
    Args:
        file_path: Path to the file
        
    Returns:
        True if file exists, False otherwise
    """
    return os.path.isfile(file_path)


def delete_file(file_path: str) -> bool:
    """
    Delete a file if it exists
    
    Args:
        file_path: Path to the file
        
    Returns:
        True if file was deleted, False otherwise
    """
    if os.path.isfile(file_path):
        os.remove(file_path)
        return True
    return False


def get_file_size(file_path: str) -> int:
    """
    Get the size of a file in bytes
    
    Args:
        file_path: Path to the file
        
    Returns:
        File size in bytes
    """
    return os.path.getsize(file_path)


def get_file_extension(file_path: str) -> str:
    """
    Get file extension from file path
    
    Args:
        file_path: Path to the file
        
    Returns:
        File extension (without dot)
    """
    return os.path.splitext(file_path)[1].lstrip('.')


def get_file_name_without_extension(file_path: str) -> str:
    """
    Get file name without extension
    
    Args:
        file_path: Path to the file
        
    Returns:
        File name without extension
    """
    return os.path.splitext(os.path.basename(file_path))[0]


def load_csv_to_dataframe(file_path: str) -> DataFrame:
    """
    Load a CSV file into a pandas DataFrame
    
    Args:
        file_path: Path to the CSV file
        
    Returns:
        pandas DataFrame containing the CSV data
    """
    return pd.read_csv(file_path)

File: typescript/fileOperations.ts
/**
 * File I/O utility functions for test automation
 */

import * as fs from 'fs';
import * as path from 'path';

/**
 * Read a text file and return its contents
 * @param filePath - Path to the text file
 * @returns File contents as a string
 */
export function readTextFile(filePath: string): string {
    return fs.readFileSync(filePath, 'utf-8');
}

/**
 * Read a text file and return its contents as an array of lines
 * @param filePath - Path to the text file
 * @returns Array of lines from the file
 */
export function readTextFileLines(filePath: string): string[] {
    const content = fs.readFileSync(filePath, 'utf-8');
    return content.split('\n');
}

/**
 * Write content to a text file
 * @param filePath - Path to the text file
 * @param content - Content to write
 */
export function writeTextFile(filePath: string, content: string): void {
    fs.writeFileSync(filePath, content, 'utf-8');
}

/**
 * Append content to a text file
 * @param filePath - Path to the text file
 * @param content - Content to append
 */
export function appendToTextFile(filePath: string, content: string): void {
    fs.appendFileSync(filePath, content, 'utf-8');
}

/**
 * Check if a file exists
 * @param filePath - Path to the file
 * @returns true if file exists, false otherwise
 */
export function fileExists(filePath: string): boolean {
    return fs.existsSync(filePath) && fs.statSync(filePath).isFile();
}

/**
 * Delete a file if it exists
 * @param filePath - Path to the file
 * @returns true if file was deleted, false otherwise
 */
export function deleteFile(filePath: string): boolean {
    if (fs.existsSync(filePath)) {
        fs.unlinkSync(filePath);
        return true;
    }
    return false;
}

/**
 * Get the size of a file in bytes
 * @param filePath - Path to the file
 * @returns File size in bytes
 */
export function getFileSize(filePath: string): number {
    const stats = fs.statSync(filePath);
    return stats.size;
}

/**
 * Get file extension from file path
 * @param filePath - Path to the file
 * @returns File extension (without dot)
 */
export function getFileExtension(filePath: string): string {
    return path.extname(filePath).substring(1);
}

/**
 * Get file name without extension
 * @param filePath - Path to the file
 * @returns File name without extension
 */
export function getFileNameWithoutExtension(filePath: string): string {
    return path.basename(filePath, path.extname(filePath));
}
